lang/java/avro/src/main/java/org/apache/avro/file/DataFileWriter.java

```diff
@@ -190,11 +190,17 @@ public boolean isFlushOnEveryBlock() {
 
   /** Open a writer appending to an existing file. */
   public DataFileWriter<D> appendTo(File file) throws IOException {
-    return appendTo(new SeekableFileInput(file),
-                    new SyncableFileOutputStream(file, true));
+    SeekableInput input = new SeekableFileInput(file);
+    OutputStream output = new SyncableFileOutputStream(file, true);
+    try {
+      return appendTo(input, output);
+    } finally {
+      input.close();
+    }
   }
 
   /** Open a writer appending to an existing file.
+   * <strong>Since 1.9.0 this method does not close in.</strong>
```

Comment:
Yes, unfortunately, this changes didn't make 1.9.0. And yes, this is an API breaking change so the next target is 1.10.0. Will rewrite the comment.